test_scripts/API/SetAppIcon/commonIconResumed.lua

```diff
@@ -0,0 +1,188 @@
+---------------------------------------------------------------------------------------------------
+-- Common module
+---------------------------------------------------------------------------------------------------
+---------------------------------------------------------------------------------------------------
+--[[ General configuration parameters ]]
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local actions = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local test = require("user_modules/dummy_connecttest")
+local commonPreconditions = require('user_modules/shared_testcases/commonPreconditions')
+local mobile_session = require('mobile_session')
+
+--[[ Module ]]
+local m = actions
+
+--[[ Variables ]]
+local hmiAppIds = {}
+
```

Comment:
@jacobkeeler , `preconditions` and `postconditions` functions are defined in `actions` and used in scripts through `common.preconditions` and `common.postconditions`.
So there is no reason to add these functions in this common file.

Reply:
Alright, looks like these functions are actually defined, my mistake.

These tests still do seem to fail because of improper cleanup, however. They pass the first time one of these tests are run and fail every subsequent time (not all, but several tests do this, including 002 and 003):

```
jacobkeeler@ubuntu:~/sdl_atf$ ./start.sh test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua 
==============================
Start 'test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua'
==============================
SDL build option EXTENDED_POLICY is unavailable.
Assume that SDL was built with EXTENDED_POLICY = PROPRIETARY
SDL build option REMOTE_CONTROL is unavailable.
Assume that SDL was built with REMOTE_CONTROL = ON
=====================================================================================================
Description: ATF test script
Severity: Major
=====================================================================================================
--------------------------------------------Preconditions--------------------------------------------
kill: (14818): No such process
[14:19:23,779] Clean_environment                                                                     [SUCCESS] (2 ms)
SDL pid 14829
tcp        0      0 127.0.0.1:8087          0.0.0.0:*               LISTEN     
HMI initialized
HMI is ready
Mobile connected
[14:19:24,809] Start_SDL__HMI__connect_Mobile__start_Session                                         [SUCCESS] (449 ms)
------------------------------------------------Test-------------------------------------------------
[14:19:25,282] App_registration_with_iconResumed___false                                             [SUCCESS] (202 ms)
[14:19:25,488] Upload_icon_file                                                                      [SUCCESS] (53 ms)
[14:19:25,543] SetAppIcon                                                                            [SUCCESS] (17 ms)
Disconnected!!!
[14:19:25,561] Disconnect_mobile_app                                                                 [SUCCESS] (22 ms)
[14:19:25,584] Connect_mobile_app                                                                    [SUCCESS] (11 ms)
[14:19:25,600] App_registration_with_iconResumed___true                                              [SUCCESS] (217 ms)
-------------------------------------------Postconditions--------------------------------------------
[14:19:25,819] Stop_SDL                                                                              [SUCCESS] (2 ms)
Total executing time is 4s 120ms (summary 4120ms)
==============================
Finish 'test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua'
==============================
jacobkeeler@ubuntu:~/sdl_atf$ ./start.sh test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua 
==============================
Start 'test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua'
==============================
SDL build option EXTENDED_POLICY is unavailable.
Assume that SDL was built with EXTENDED_POLICY = PROPRIETARY
SDL build option REMOTE_CONTROL is unavailable.
Assume that SDL was built with REMOTE_CONTROL = ON
=====================================================================================================
Description: ATF test script
Severity: Major
=====================================================================================================
--------------------------------------------Preconditions--------------------------------------------
kill: (15296): No such process
[14:19:28,898] Clean_environment                                                                     [SUCCESS] (2 ms)
SDL pid 15307
tcp        0      0 127.0.0.1:8087          0.0.0.0:*               LISTEN     
HMI initialized
HMI is ready
Mobile connected
[14:19:29,926] Start_SDL__HMI__connect_Mobile__start_Session                                         [SUCCESS] (437 ms)
------------------------------------------------Test-------------------------------------------------
[14:19:30,386] App_registration_with_iconResumed___false                                             [FAIL] (197 ms)
 HMI notification BasicCommunication.OnAppRegistered: ValidIf: 
BC.OnAppRegistered notification contains unexpected parameter: icon 
 response to 2: ValidIf: 
payload.iconResumed: expected: false, actual value: true
[14:19:30,587] Upload_icon_file                                                                      [SUCCESS] (51 ms)
[14:19:30,639] SetAppIcon                                                                            [SUCCESS] (19 ms)
Disconnected!!!
[14:19:30,662] Disconnect_mobile_app                                                                 [SUCCESS] (17 ms)
[14:19:30,681] Connect_mobile_app                                                                    [SUCCESS] (9 ms)
[14:19:30,694] App_registration_with_iconResumed___true                                              [SUCCESS] (201 ms)
-------------------------------------------Postconditions--------------------------------------------
QIODevice::read: device not open
[14:19:30,897] Stop_SDL                                                                              [SUCCESS] (2 ms)
Total executing time is 4s 71ms (summary 4071ms)
==============================
Finish 'test_scripts/API/SetAppIcon/003_Icon_resumption_by_disconnect.lua'
==============================
```